internal/mempool/mempool_test.go

```diff
@@ -2549,3 +2549,130 @@ func TestHandlesTSpends(t *testing.T) {
 		t.Fatalf("OnTSpendReceived callback was not called")
 	}
 }
+
+// TestHandlesTAdds verifies that the mempool correctly adds and removes valid
+// tadds.
+func TestHandlesTAdds(t *testing.T) {
+	t.Parallel()
+
+	// TODO: switch to mainnet params once they have treasury params set so
+	// we can more accurately test the behavior of the mempool under
+	// production parameters.
+	net := chaincfg.SimNetParams()
+	harness, outs, err := newPoolHarness(net)
+	if err != nil {
+		t.Fatalf("unable to create test pool: %v", err)
+	}
+	tc := &testContext{t, harness}
+
+	// Setup the harness for the test and activate the treasury agenda.
+	harness.txPool.cfg.IsTreasuryAgendaActive = func() (bool, error) {
+		return true, nil
+	}
+	harness.txPool.cfg.Policy.MaxTxVersion = 3
+	harness.chain.SetStandardVerifyFlags(BaseStandardVerifyFlags | txscript.ScriptVerifyTreasury)
+
+	// Addresses to use in the following TAdds.
+	addPrivKey := secp256k1.NewPrivateKey(new(secp256k1.ModNScalar).SetInt(2))
+	pubKey := addPrivKey.PubKey().SerializeCompressed()
+	pubKeyHash := dcrutil.Hash160(pubKey)
+	addP2pkhAddr, err := dcrutil.NewAddressPubKeyHash(pubKeyHash, net,
+		dcrec.STEcdsaSecp256k1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	addP2shScript := []byte{txscript.OP_NOP, txscript.OP_NOP, txscript.OP_TRUE}
+	addP2shAddr, err := dcrutil.NewAddressScriptHash(addP2shScript, net)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Helper to generate TAdds as needed.
+	createTAdd := func(spend *spendableOutput,
```

Comment:
Harkening back to an earlier comment I made on a previous commit about how I think the ability to create these should be separate since it's useful in more than one test.